librz/arch/isa/mips/il/common.c

```diff
@@ -0,0 +1,1024 @@
+// SPDX-FileCopyrightText: 2025 deroad <deroad@kumo.xn--q9jyb4c>
+// SPDX-FileCopyrightText: 2023 Siddharth Mishra <admin@brightprogrammer.in>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#define MIPS_DWORD_SIZE 64
+#define MIPS_WORD_SIZE  32
+#define MIPS_HALF_SIZE  16
+#define MIPS_BYTE_SIZE  8
+
+#define MIPS_REG_LO "lo"
+#define MIPS_REG_HI "hi"
+#define MIPS_REG_RA "ra"
+
+#define IS_ZERO_REG(idx) mips_is_zero_reg(insn, idx)
+#define TRUNC32(x)       UNSIGNED(MIPS_WORD_SIZE, x)
+#define TRUNC16(x)       UNSIGNED(MIPS_HALF_SIZE, x)
+#define TRUNC8(x)        UNSIGNED(MIPS_BYTE_SIZE, x)
+
+#define BITN(x, n)            SHIFTR0(LOGAND(x, UN(gprlen, (ut64)1 << (n - 1))), UN(gprlen, n - 1))
+#define CHECK_OVERFLOW(r, sz) EQ(BITN(r, sz), BITN(r, sz - 1))
+#define MIPS_REG(idx)         mips_get_reg(handle, insn, idx, gprlen)
+#define MIPS_IMM(idx)         UN(gprlen, IMM(idx))
+#define MIPS_ZERO()           UN(gprlen, 0)
+#define MIPS_LINK()           SETG(MIPS_REG_RA, UN(gprlen, insn->address + 8)) // link register $ra
```

Comment:
```suggestion
#define MIPS_LINK()           SETG(MIPS_REG_RA, UN(gprlen, insn->address + 8)) /* link register $ra */
```

In case people use funny per-processors